src/Codec.elm

```diff
@@ -442,6 +452,54 @@ nullableField name getter codec ocodec =
     field name getter (maybe codec) ocodec
 
 
+{-| Basically a `maybeField` where data is validated when provided and a default value to use when not.
+-}
+optionalField : String -> (a -> Maybe f) -> Codec f -> ObjectCodec a (Maybe f -> b) -> ObjectCodec a b
+optionalField name getter codec (ObjectCodec ocodec) =
+    ObjectCodec
+        { encoder =
+            \v ->
+                case getter v of
+                    Just present ->
+                        ( name, encoder codec present ) :: ocodec.encoder v
+
+                    Nothing ->
+                        ocodec.encoder v
+        , decoder =
+            decodeOptionalField name codec
+                |> JD.map2 (\f x -> f x) ocodec.decoder
+        }
+
+
+{-| Basically an `optionalField` where you can provide a default value in case data is missing.
```

Comment:
How about (ter):
```
{- | Similar to the `optionalField` codec, but the default value is provided in case the data is missing.
    This means that bad data will generate a decoding error, instead of silently failing and returning the default value.

    - valid data provided: returns the decoded value
    - no data provided (either the field is missing, or it is null): returns the default value
    - invalid data provided: returns a decoding error

    type alias JustAnInt = { myInt: Int }

    objectCodec = Codec.object JustAnInt
        |> Codec.optionalFieldWithDefault "myInt" .myInt Codec.int 0
        |> Codec.buildObject

    Codec.decodeString objectCodec "{\"myInt\": \"Not an int\"}"
    --Returns a json decoding error with the message containing: "Expecting an INT"
    
    Codec.decodeString objectCodec "{\"myInt\": null}"
    --Returns: Ok { myInt = 0 } 
    
    Codec.decodeString objectCodec "{}"
    --Returns: Ok { myInt = 0 } 
-}
```